File: Dart/CNDARTLANG/bin/Functions/Newton.dart
import 'dart:math';

fx(double x) => (pow(x,4))+2*pow(x,3)-13*pow(x,2)-14*x+24;
dfx(double x) => 4*pow(x,3)+6*pow(x,2)-26*x-14;

Newton(double x0, double Toler, int IterMax) {
  double Fx = fx(x0);
  double DFx = dfx(x0);
  double x = x0;
  double Iter = 0;
  double DeltaX = 1000;
  double DF = 1;
  double CondErro;

  while (((DeltaX).abs() > Toler || (Fx).abs() > Toler) && (DF != 0) &&
      (Iter < IterMax)){
    DeltaX = -Fx / DFx;
    x = x + DeltaX;
    Fx = fx(x);
    DFx = dfx(x);
    Iter = Iter + 1;
  }
  double Raiz = x;
  if ((DeltaX).abs() <= Toler && (Fx).abs() <= Toler) {
    CondErro = 0;
  }
  else {
    CondErro = 1;
  }
  return "Raiz = " + Raiz.toString() + ',\n' + "Iter = " + Iter.toString() + ',\n' + "CondErro = " + CondErro.toString();
}

File: JavaScript/newton.js
console.time('tempo');
x0 = 4;
toler = 10000*Math.E-5;
itermax = 100;
newton(x0,toler,itermax);
console.timeEnd('tempo');

//8 Método de Newton
function newton(x0,toler,itermax){
    fx = Math.pow(x0,4)+2*Math.pow(x0,3)-13*Math.pow(x0,2)-14*x0+24;
    dfx = 4*Math.pow(x0,3)+6*Math.pow(x0,2)-26*x0-14;

    x= x0;
    iter = 0;
    deltax=1000;
    df=1;
    while((Math.abs(deltax)>toler || Math.abs(fx) > toler) && df!==0 && iter < itermax){
        deltax=-fx/dfx;
        x=x+deltax;

        fx = Math.pow(x0,4)+2*Math.pow(x0,3)-13*Math.pow(x0,2)-14*x0+24;
        dfx = 4*Math.pow(x0,3)+6*Math.pow(x0,2)-26*x0-14;

        iter=iter+1;
    }

    raiz = x;
    if(Math.abs(deltax)<=toler && Math.abs(fx)<=toler){
        conderro = 0;
    }else{
        conderro = 1;
    }

    document.write("Método de Newton<br>");
    document.write("Iter: "+iter+"<br>");
    document.write("raiz: "+raiz+"<br>");
    document.write("condErro: "+conderro+"<br>");
}




